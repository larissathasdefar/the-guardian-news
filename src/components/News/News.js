import React from 'react'
import _ from 'prop-types'
import styled from 'styled-components'
import { format } from 'date-fns'
import Card from '@material-ui/core/Card'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'

const Wrapper = styled(Card)`
    margin-bottom: 6px;
    cursor: pointer;

    &:hover {
        background-color: #eeeeee;
    }
`

const Actions = styled(CardActions)`
    justify-content: space-between;
`

const News = ({ title, section, date }) => (
    <Wrapper data-testid="news-content">
        <CardContent>
            <Typography gutterBottom>
                {`[${section}] ${title}`}
            </Typography>
        </CardContent>
        <Actions>
            <Button size="small">Learn More</Button>
            <Typography variant="caption">{format(new Date(date), 'dd/MM/yyyy HH:mm:ss')}</Typography>
        </Actions>
    </Wrapper>
)

News.propTypes = {
    title: _.string.isRequired,
    section: _.string.isRequired,
    date: _.string.isRequired,
}

export default News;
