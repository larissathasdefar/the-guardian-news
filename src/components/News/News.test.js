import React from 'react'
import { render } from '@testing-library/react'
import News from './News'

const props = {
    section: 'Football',
    date: "2020-02-25T19:00:35Z",
    title: "Real Madrid and ban give Pep Guardiola highest-stakes test yet",
}

describe('News component', () => {
    test('Renders News component', async () => {
        const { findByTestId } = render(<News {...props} />)
        const component = await findByTestId('news-content')

        expect(component).toHaveTextContent(props.title)
        expect(component).toHaveTextContent('25/02/2020 16:00:35')
    })

    test('Snapshot', () => {
        const { asFragment } = render(<News {...props} />)
        expect(asFragment()).toMatchSnapshot()
    })

})
