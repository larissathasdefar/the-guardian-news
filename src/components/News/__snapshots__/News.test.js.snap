// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root sc-AxjAm hSVQxc MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="news-content"
  >
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        [Football] Real Madrid and ban give Pep Guardiola highest-stakes test yet
      </p>
    </div>
    <div
      class="MuiCardActions-root sc-AxirZ dPyOpm MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Learn More
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <span
        class="MuiTypography-root MuiTypography-caption"
      >
        25/02/2020 16:00:35
      </span>
    </div>
  </div>
</DocumentFragment>
`;
